Makefile

```diff
@@ -7,7 +7,12 @@ endif
 
 BUILDDIR=build
 
-all: awesome ;
+# Run "make" in $(BUILDDIR) by default.
+# This is required to generate all files already, which should not be generated
+# with "(sudo) make install" only later.
+cmake-build: $(BUILDDIR)/Makefile
+	$(ECHO) "Building…"
+	$(MAKE) -C $(BUILDDIR)
```

Comment:
isn't it running `$(MAKE) -C $(BUILDDIR)` twice?

Reply:
In which case?

Just `make` defaults to `make cmake-build` now, which runs `make -C build`.
`make foo` will run `make -C build foo` (due to the implicit rule at the end).

The problem before 0b36da273 was that the default was `all`, which depended on `awesome`, which depended on `cmake-build`.  So cmake-build was run (`make -C build`), then `make -C build awesome` again.

Reply:
`cmake-build` build target depends on `$(BUILDDIR)/Makefile`

and both of them are running `$(MAKE) -C $(BUILDDIR)` one of them just bypassing args at the end (`$@`)

Reply:
`make` (or `make cmake-build`) does not also build the `%` target.

Reply:
`%` is only used if not other target is specified (including the default).